example/src/App.tsx

```diff
@@ -6,28 +6,30 @@ import {
   StyleSheet,
   View,
 } from 'react-native';
-import TrackPlayer from 'react-native-track-player';
+import TrackPlayer, { useActiveTrack } from 'react-native-track-player';
 
 import { Button, PlayerControls, Progress, TrackInfo } from './components';
-import { useCurrentTrack } from './hooks';
 import { QueueInitialTracksService, SetupService } from './services';
 
 const App: React.FC = () => {
-  const track = useCurrentTrack();
+  const track = useActiveTrack();
   const [isPlayerReady, setIsPlayerReady] = useState<boolean>(false);
 
   useEffect(() => {
-    async function run() {
+    let unmounted = false;
+    (async () => {
```

Comment:
This IIFE is nice. But I reckon it will be confusing to those less experienced or knowledgable in JS. I can imagine people saying to themselves "What the heck is happening her??"

I reckon demos/examples should always be super easy and cater to less level developers where possible.